Source/Frontend/UI/UI_VanguardImplementation.cs

```diff
@@ -472,5 +150,453 @@ private static void OnMessageReceived(object sender, NetCoreEventArgs e)
+            });
+        }
+
+        private static void DisableRealTimeSupport()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                Button btnManual = S.GET<UI_CoreForm>().btnManualBlast;
+                if (AllSpec.VanguardSpec[VSPEC.REPLACE_MANUALBLAST_WITH_GHCORRUPT] != null)
+                {
+                    btnManual.Text = "  Corrupt";
+                }
+                else
+                {
+                    btnManual.Visible = false;
+                }
+
+                S.GET<UI_CoreForm>().btnAutoCorrupt.Enabled = false;
+                S.GET<UI_CoreForm>().btnAutoCorrupt.Visible = false;
+                S.GET<RTC_GeneralParameters_Form>().multiTB_ErrorDelay.Enabled = false;
+                S.GET<RTC_GlitchHarvesterBlast_Form>().btnSendRaw.Enabled = false;
+                S.GET<RTC_GlitchHarvesterBlast_Form>().btnBlastToggle.Enabled = false;
+
+                S.GET<RTC_CorruptionEngine_Form>().cbSelectedEngine.Items.Remove("Hellgenie Engine");
+                S.GET<RTC_CorruptionEngine_Form>().cbSelectedEngine.Items.Remove("Distortion Engine");
+                S.GET<RTC_CorruptionEngine_Form>().cbSelectedEngine.Items.Remove("Pipe Engine");
+                S.GET<RTC_CorruptionEngine_Form>().cbSelectedEngine.Items.Remove("Freeze Engine");
+            });
+        }
+
+        private static void DisableKillSwitchSupport()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                S.GET<UI_CoreForm>().pnAutoKillSwitch.Visible = false;
+                S.GET<UI_CoreForm>().cbUseAutoKillSwitch.Checked = false;
+            });
+        }
+
+        private static void StartSanitizeTool()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                var blastEditor = S.GET<RTC_NewBlastEditor_Form>();
+                blastEditor.OpenSanitizeTool(false);
+            });
+        }
+
+        private static void LoadCorrupt()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                var blastEditor = S.GET<RTC_NewBlastEditor_Form>();
+                blastEditor.btnLoadCorrupt_Click(null, null);
+            });
+        }
+
+        private static void LoadOriginal()
+        {
+            SyncObjectSingleton.FormExecute(() =>
+            {
+                var blastEditor = S.GET<RTC_NewBlastEditor_Form>();
+                blastEditor.LoadOriginal();
+            });
+        }
+
+        private static void GetLayerSizeUnlockedUnits(ref NetCoreEventArgs e)
+        {
+            var units = 0;
+            SyncObjectSingleton.FormExecute(() =>
+            {   // this is what the sanitize tool uses to judge how many units there are left to sanitize.
+                var blastEditor = S.GET<RTC_NewBlastEditor_Form>();
+                units = blastEditor.currentSK?.BlastLayer?.Layer.Count(x => !x.IsLocked) ?? -1;
```

Comment:
I had to modify this function since `ref NetCoreEventArgs e` can't be used inside of an `Action`

Reply:
This is fine for now. We can re-evaluate how we want to handle the return values for future. We have a concept to rework the massive switch statement into something more clean/maintainable (codegen from attributes?) so when we address that, it'll replace this.